src/MoBi.UI/Views/RenameExpressionProfileBuildingBlockView.cs

```diff
@@ -0,0 +1,75 @@
+﻿using DevExpress.XtraEditors;
+using MoBi.Presentation.DTO;
+using MoBi.Presentation.Presenter;
+using OSPSuite.Assets;
+using OSPSuite.DataBinding;
+using OSPSuite.DataBinding.DevExpress;
+using OSPSuite.UI.Views;
+
+namespace MoBi.UI.Views
+{
+   public partial class NewNameExpressionProfileBuildingBlockView : BaseModalView, INewNameExpressionProfileBuildingBlockView
+   {
+      private readonly ScreenBinder<RenameExpressionProfileDTO> _screenBinder = new ScreenBinder<RenameExpressionProfileDTO>();
+      public NewNameExpressionProfileBuildingBlockView()
+      {
+         InitializeComponent();
+         ExtraVisible = false;
+         tbName.ReadOnly = true;
+         MaximizeBox = false;
+         MinimizeBox = false;
+
+         initializeResources();
+         initializeBinding();
+      }
+
+      public override bool HasError => _screenBinder.HasError;
+
+      private void initializeBinding()
+      {
+         _screenBinder.Bind(x => x.Category).To(tbCategory);
+         _screenBinder.Bind(x => x.MoleculeName).To(tbMoleculeName);
+         _screenBinder.Bind(x => x.Species).To(tbSpecies);
+         _screenBinder.Bind(x => x.Name).To(tbName);
+         _screenBinder.SavingMode = SavingMode.Always;
+
+         // We need to validate the text box when the text changes so that the Name property is validated
+         // even thought that text box is readonly. The Name is composed of each of the other text boxes content
+         tbCategory.TextChanged += (o,e) => OnEvent(() => validateTextBox(o));
```

Comment:
I don't get this. why is this required? Isn't it what the property changes is all about?

Reply:
That's only updated when the field loses focus. To get this feature working the way I liked, I wanted to run validation on every keystroke.

Reply:
I should say that the validation for that field already runs on each keystroke, but the underlying field is only updated on lost focus.

I want to update the underlying field and validate the 'Name' property on every keystroke on any one of these text boxes.

Reply:
This should work out of the box. Not a big deal but if you have your DTO raising the right changed event...it happens automatically like for all other boxes.